src/actions/loopring/index.js

```diff
@@ -15,8 +15,84 @@ import { getDepth } from "../../lightcone/api/v1/depth/get";
 import { getMarketInfo } from "../../lightcone/api/v1/marketinfo/get";
 import config from "../../lightcone/config";
 import { getBalances } from "../../lightcone/api/v1/balances";
-import { fromWei } from "web3-utils";
 import { getFeeRates } from "../../lightcone/api/v1/fee-rates";
+import { weiToEther } from "../../utils";
+import { getEthNonce } from "../../lightcone/api/v1/ethnonce/get";
+import { getRecommendedGasPrice } from "../../lightcone/api/v1/recommendedGasPrice/get";
+
+// auth status
+
+export const GET_AUTH_STATUS_SUCCESS = "GET_AUTH_STATUS_SUCCESS";
+export const GET_AUTH_STATUS_START = "GET_AUTH_STATUS_START";
+export const GET_AUTH_STATUS_END = "GET_AUTH_STATUS_END";
+
+export const getAuthStatus = (address) => async (dispatch) => {
+    try {
+        dispatch({ type: GET_AUTH_STATUS_START });
+        let needsRegistration = false;
+        try {
+            await lightconeGetAccount(address);
+        } catch (error) {
+            needsRegistration = true;
+        }
+        dispatch({ type: GET_AUTH_STATUS_SUCCESS, needsRegistration });
+    } catch (error) {
+        toast.error(<FormattedMessage id="error.auth.status" />);
+        console.error("error checking auth status", error);
+    } finally {
+        dispatch({ type: GET_AUTH_STATUS_END });
+    }
+};
+
+// registration
+
+export const register = (web3Instance, ethereumAccount) => async (dispatch) => {
+    try {
+        const wallet = new Wallet("MetaMask", web3Instance, ethereumAccount);
```

Comment:
Is it only for MetaMask?

Reply:
In this instance I copied the code that I've seen in the `dexwebapp` repo. I don't know how that applies to other wallets, it might be outdated (if updates on `dexwebapp` have been done).

Reply:
Please https://github.com/Loopring/dexwebapp/search?q=new+Wallet&unscoped_q=new+Wallet 

Different wallets use different values.

Reply:
Oh sorry, didn't see it at first. I will fix it later today. What about merging and releasing this version (which has fixes on the slippage calculation and on the display of the balances, on which people have actually complained)? 
I'll then I'll open a PR with the fixes to the wallet instance creation later.
Because my guess is that I won't have time to do it until this evening (i.e. in 8 hrs approximately), and I've seen multiple people actually complaining about the slippage etc.

Reply:
You can remove the code about the registration from the PR. 

I think mixing two features or two issues into one PR is not a good software practice. It's making noises.

Reply:
As far as the Metamask problem, I had a look at the code. Turns out the wallet type is only used in managing the local storage.
I'm currently not using this feature. It's code I inherited by copying the `dexwebapp`'s `src/lightcone` folder into the Swap UI, but yeah, I'm not using it.
As far as opening another PR, nope I'm sorry. I won't remove the registration feature from here just to open another PR with it. It's already here, why waste any time? People are asking for it along with the fixes contained in this PR.

Reply:
I don't think it waste time. 

If you are not using the feature, why not remove it in the current PR?

Reply:
I don't think I understand. The registration feature is here, and it will indeed be used, along with the other micro, micro fixes people are asking for.
The slippage fix is urgent. Right now people look at the wrongly calculated slippage and are scared away off the platform.
They are also confused about the lack of a registration option.
These are the two things that they are looking for right now.

>If you are not using the feature, why not remove it in the current PR?

The goal of the current PR is to be merged urgently to fix some issues getting in the way of the users. Users do not care about dead code being here I think. So my suggestion is: could we merge this now and then I'll fix the other things (i.e. remove the dead code) in a subsequent PR?

Reply:
I will merge the PR and remove the code myself this time.

Security is our first priority all the time. Any code related to account registration and keys requires us to pay more attentions. Dead code will hide security issues.

Reply:
It's the exact same code that you use in the dex webapp though. But I see where you're coming from

Reply:
It's not the same code. dexwebapp has supported different wallets.